lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVMPass.cpp

```diff
@@ -310,6 +312,33 @@ class ConvertTritonGPUToLLVM
     });
   }
 
+    void decomposeMmaToMmaOperand(ModuleOp mod, int numWarps) const {
+    // Replace `mma -> mma` with `mma -> blocked -> mma`
+    // unless certain conditions are met
+    mod.walk([&](triton::gpu::ConvertLayoutOp cvtOp) -> void {
+      OpBuilder builder(cvtOp);
+      auto srcType = cvtOp.getOperand().getType().cast<RankedTensorType>();
+      auto dstType = cvtOp.getType().cast<RankedTensorType>();
+      auto srcMma =
+          srcType.getEncoding().dyn_cast<triton::gpu::MmaEncodingAttr>();
+      auto dstMma =
+          dstType.getEncoding().dyn_cast<triton::gpu::MmaEncodingAttr>();
+      if (srcMma && dstMma) {
+        auto tmpType = RankedTensorType::get(
+            dstType.getShape(), dstType.getElementType(),
+            triton::gpu::BlockedEncodingAttr::get(
+                mod.getContext(), srcType.getShape(), getSizePerThread(srcMma),
+                getOrder(srcMma), numWarps));
+        auto toBlock = builder.create<triton::gpu::ConvertLayoutOp>(
+            cvtOp.getLoc(), tmpType, cvtOp.getOperand());
+        auto toMma  = builder.create<triton::gpu::ConvertLayoutOp>(
+            cvtOp.getLoc(), dstType, toBlock);
+        cvtOp.replaceAllUsesWith(toMma.getResult());
+        cvtOp.erase();
+      }
+    });
+  }
+
```

Comment:
This part transforms one `mma -> mma` layout conversion to `mma -> block -> mma` conversions to utilize existing conversions